cmd/binlogctl/main.go

```diff
@@ -41,25 +42,32 @@ func main() {
 		os.Exit(2)
 	}
 
+	if cfg.SSLCA != "" {
+		err = binlogctl.InitHTTPSClient(cfg.SSLCA, cfg.SSLCert, cfg.SSLKey)
+		if err != nil {
+			log.Fatal("failed to init https client", zap.Error(err))
+		}
+	}
+
 	switch cfg.Command {
 	case ctl.GenerateMeta:
 		err = ctl.GenerateMetaInfo(cfg)
 	case ctl.QueryPumps:
-		err = ctl.QueryNodesByKind(cfg.EtcdURLs, node.PumpNode, cfg.ShowOfflineNodes)
+		err = ctl.QueryNodesByKind(cfg.EtcdURLs, node.PumpNode, cfg.ShowOfflineNodes, cfg.TLS)
```

Comment:
I think we should initial all clients (like `registry`, `dial Client`) together, the reason is to ensure that `TLS` are processed correctly and uniformly, here implementation is a weird example:

* we provide `TLS` config in interface of `github.com/pingcap/tidb-binlog/binlogctl`, like `QueryNodesByKind `
* we do `binlogctl.InitHTTPSClient` separately

Reply:
https://github.com/pingcap/tidb-binlog/pull/904/commits/0ab7edb6dcde3ff161367cda1d9e93209d9da193